lib/ansible/modules/net_tools/basics/uri.py

```diff
@@ -317,6 +336,38 @@ def absolute_location(url, location):
         return location
 
 
+def kv_list(data):
+    ''' Convert data into a list of key-value tuples '''
+    if data is None:
+        return None
+
+    if isinstance(data, Sequence):
+        return list(data)
+
+    if isinstance(data, Mapping):
+        return list(data.items())
+
+    raise ValueError('cannot form-urlencode body, expect list or dict')
+
+
+def form_urlencoded(body):
```

Comment:
There was a question posted yesterday as to why this function is necessary.  After staring at this an playing with it, it seems to be because from ansible, the example of:

```
body:
      - [ user, foo ]
      - [ password, bar!#@ |&82$M ]
      - [ submit, Sign in ]
```
 Would become:
```
[['user', 'foo'], ['password', 'bar!#@ |&82$M'], ['submit', 'Sign in']]
```

Which `urlencode` cannot accept, as it expects a list of tuples in this case.

Please add documentation as to why this function needs to exist, so that we understand why you just didn't use `urlencode` directly.

Reply:
Right, I will add a comment.